content/articles/setup-and-run-applications-using-typescript-with-aureliajs/index.md

```diff
@@ -0,0 +1,270 @@
+
+- Add a filter component
+
+First, add a filter component. This will be sued to filter any todo from ye list of the added todos.
+
+```html
+<nav class="navbar navbar-daNrk bg-dark">
+    <div class="container-fluid">
+        <a class="navbar-brand">Todo App</a>
+        <form class="d-flex">
+            <input class="form-control me-2" type="search" placeholder="Search" aria-label="Search">
+            <button class="btn btn-outline-success" type="submit">Search</button>
+        </form>
+    </div>
+</nav>
+```
+
+- Get the application header
+
+```html
+<h1 class="text-center">${heading}</h1>
+```
+
+This header is defined inside the `app.ts` inside the class `App`. `heading` is a property of the model. The above line of code will bind it to the template using the string interpolation. The `${}` operator helps you extract the `heading` string value.
+
+- Add a form to add todos
+
+```html
+<form submit.trigger="addTodo()">
+    <label for="addTodoInput" class="form-label">Enter New Todo</label>
+</form>
+```
+
+This form will trigger the constructor `addTodo()` to manipulate the data.
+
+- Add todo title form input
+
+```html
+<input type="text" value.bind="todoTitle" id="addTodoInput" class="form-control" placeholder="Enter Title">
+```
+
+Like the model states, each todos has three properties: a title, description, and completed value. In this case, the completed value is added to false by default. The above code block will provide a form to enter the todo title.
+
+- Add todo description form input
+
+```html
+<input type="text" value.bind="todoDescription" id="addTodoInput" class="form-control" placeholder="Enter Description">
+```
+
+- Add a button to submit the form input
+
+```html
+<button class="btn btn-primary" type="submit" disabled.bind="!todoTitle">Submit</button>
+
+<div class="form-check" repeat.for="todo of todos">
+</div>
+```
+
+To link the view model's `todoTitle` and `todoDescription`, use the `value.bind`. This will then map the value to the template.
+
+- Add a checkbox
+
+```html
+<input class="form-check-input" type="checkbox" value="" id="flexCheckDefault" checked.bind="todo.completed">
+```
+
+This will be used to update the todos using the set `completed` value.
+
+- Update a todo
+
+```html
+<label class="form-check-label" for="flexCheckDefault"
+    css="text-decoration: ${todo.completed ? 'line-through' : 'none'}">
+    ${todo.title}
+    ${todo.description}
+</label>
+```
+
+This will access the todos description and title. When the checkbox is clicked, Aurelia will check the `todo.completed` value and add line-through to make the todo as completed.
+
+- Delete a todo
+
+```html
+<button class="btn btn-danger" click.trigger="removeTodo(todo)">Delete</button>
+```
+
+- Clone a todo
+
+```html
+<button click.trigger="cloneTodo(todo)">Clone</button>
+```
+
+Finally, run `npm start` and open `http://localhost:8080/` on the browser to start interacting with your Aurelia application.
+
```

Comment:
This looks like an abrupt end.

Add necessary images the demonstrates the work

Reply:
If possible, add images that demo's each stage of your work.

Like adding a todo, cloning a todo and so on